pkg/fanal/analyzer/language/java/sbt/lockfile.go

```diff
@@ -0,0 +1,86 @@
+package sbt
+
+import (
+	"context"
+	"io"
+	"io/fs"
+	"os"
+	"strings"
+
+	"golang.org/x/xerrors"
+
+	"github.com/aquasecurity/trivy/pkg/dependency/parser/sbt/lockfile"
+	"github.com/aquasecurity/trivy/pkg/fanal/analyzer"
+	"github.com/aquasecurity/trivy/pkg/fanal/analyzer/language"
+	"github.com/aquasecurity/trivy/pkg/fanal/types"
+	"github.com/aquasecurity/trivy/pkg/utils/fsutils"
+)
+
+func init() {
+	analyzer.RegisterPostAnalyzer(analyzer.TypeSbtLock, newSbtDependencyLockAnalyzer)
+}
+
+const (
+	version        = 1
+	fileNameSuffix = "sbt.lock"
+)
+
+// sbtDependencyLockAnalyzer analyzes '*.sbt.lock'
+type sbtDependencyLockAnalyzer struct {
+	parser language.Parser
+}
+
+func newSbtDependencyLockAnalyzer(_ analyzer.AnalyzerOptions) (analyzer.PostAnalyzer, error) {
+	return &sbtDependencyLockAnalyzer{
+		parser: lockfile.NewParser(),
+	}, nil
+}
+
+func (a sbtDependencyLockAnalyzer) PostAnalyze(_ context.Context, input analyzer.PostAnalysisInput) (*analyzer.AnalysisResult, error) {
```

Comment:
We don't need to parse other files (we only need lock file)
So I think it's better to use https://github.com/aquasecurity/trivy/blob/6e7f62d2debbb4acbeb7ad58ecd8ac7379f19e40/pkg/fanal/analyzer/analyzer.go#L71-L76

Reply:
I thought we would parse other files in the future. Let's ask the future plan of sbt-dependency-lock.

Reply:
Future improvements to sbt-dependency-lock will be added to the existing `*.sbt.lock` files.

Is it still possible to walk the directory in an `analyzer`? It is common to have multiple sub-projects in sbt and each of these would generate their own `build.sbt.lock` file.

Reply:
What is the sub-project? Does it belong to the main project? Or are they separate projects? If you need to process those files at the same time, you need to use `PostAnalyzer`. If you only need to parse them individually, `Analyzer` is more efficient. For example, Trivy needs both `package.json` and `package-lock.json` for npm projects, and uses `PostAnalyzer` in this case.

Reply:
sbt supports having multiple projects within a single parent project. Generally each sub-project will generate its own artifacts (JARs/WARs/GraalVM binaries etc) and will be in a sub-directory of the parent project.

Configuration for sub-projects (including dependencies) is _extremely_ flexible and can be defined in any of the following locations (or a combination):

- `build.sbt` in the sub-project dir.
- inherited from the root `build.sbt`.
- configured from the `project` directory of the root project.

The sbt-dependency-lock plugin will generate a separate `build.sbt.lock` file in each of the sub-projects that will contain all the dependencies of that sub-project.

Reply:
Do you want to process `build.sbt.lock` separately or together? When processed together, all dependent packages are shown as root `build.sbt.lock`. If not, they are just different lock files and show dependencies respectively.

Reply:
After thinking about it I reckon that processing them separately is probably better for the majority of sbt users.

Reply:
Thanks for your insight. We can use the analyzer for now and switch to the post-analyzer when we need to combine those files.